tiledb/api/src/array/dimension/mod.rs

```diff
@@ -206,6 +205,64 @@ impl<'ctx> Builder<'ctx> {
         })
     }
 
+    pub fn new_string(
+        context: &'ctx Context,
+        name: &str,
+        datatype: Datatype,
+    ) -> TileDBResult<Self> {
+        let c_datatype = datatype.capi_enum();
+        let c_name = cstring!(name);
+        let mut c_dimension: *mut ffi::tiledb_dimension_t =
+            std::ptr::null_mut();
+
+        context.capi_call(|ctx| unsafe {
+            ffi::tiledb_dimension_alloc(
+                ctx,
+                c_name.as_ptr(),
+                c_datatype,
+                std::ptr::null(),
+                std::ptr::null(),
+                &mut c_dimension,
+            )
+        })?;
+        Ok(Builder {
+            dim: Dimension {
+                context,
+                raw: RawDimension::Owned(c_dimension),
+            },
+        })
+    }
+
+    // This is internal only on purpose.
```

Comment:
The comment should probably include some notes from the above discussion as to what's up

Reply:
Fixed.